Modules/_decimal/_decimal.c

```diff
@@ -5898,7 +5899,12 @@ _decimal_exec(PyObject *m)
     mpd_reallocfunc = PyMem_Realloc;
     mpd_callocfunc = mpd_callocfunc_em;
     mpd_free = PyMem_Free;
-    mpd_setminalloc(_Py_DEC_MINALLOC);
+
+    /* Suppress the warning caused by multi-phase initialization */
+    if (minalloc_is_set) {
```

Comment:
Are you sure of expression in brackets? Isn't there should been `(!minalloc_is_set)`?

Reply:
Oh thanks! I made a mistake here.
Amazingly, the tests all pass even though this code is not executed once.